src/backend/access/transam/xlogfuncs.c

```diff
@@ -175,6 +180,16 @@ pg_create_restore_point(PG_FUNCTION_ARGS)
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 				 errmsg("value too long for restore point (maximum %d characters)", MAXFNAMELEN - 1)));
 
+	/*
+	 * Dispatch creation of restore point to segments
+	 * TODO: Commit blocking has to be introduced here prior the dispatch
+	 */
+	if (IS_QUERY_DISPATCHER())
+	{
+		sprintf(command, "SELECT pg_catalog.pg_create_restore_point('%s')", restore_name_str);
+		CdbDispatchCommand(command, DF_NONE, &cdb_pgresults);
+	}
+
```

Comment:
As simple as this looks, we should really be creating a new catalog function `gp_create_restore_point()` for distributed restore point.  For one thing, this function returns an LSN of the master but the function we desire will not return a result at all (or we could do a result set of all LSNs as separate records or as an array of LSNs).  In general, we should leave postgres catalog functions untouched unless the change is required.

Reply:
Considered that but didn't quite know if users would actually really use the list of LSNs when returned back as opposed to simply the master's LSN in this approach. Can introduce the `gp_create_restore_point` if you feel strongly about leaving the postgres function as is.